### ISSUE TITLE
Configuration options for server request/response validation required

### ISSUE THREAD

**Issue by jeromekelleher** [ga4gh/server#175]
The `ProtocolElement.validate()` method allows us to check whether a given JSON dictionary is a valid instance of the protocol. We should have options to enable this checking for all incoming requests and all outgoing responses. We need both because we want them for different things: validation of input will be useful in productions settings as a security measure, and validation of output will be useful for development and testing.

One way to do this would be:

Give the `backend` object a reference to the Flask `config` object. We should add a method to backend called `setConfig` which can be called (for now) in `cli.server_main`, e.g. 
```python
backend = backend.Backend(...)
frontend.app.backend = backend
backend.setConfig(frontend.app.config)
```
This does create coupling between the frontend and the backend, but hopefully by just using the config object this isn't too bad.

It is then straightforward to update `Backend.runSearchRequest()` to optionally validate the request and response.

**Comment by dcolligan** [comment]
I'm fine with this change, except for the coupling between frontend and backend.

It's probably a good idea to break out the flask config into its own module that can be then both imported by frontend and backend.

**Comment by dcolligan** [comment]
Fixed in #205
